### PR TITLE
[rbac] Refactored scope names

### PR DESCRIPTION
**Pull request by 0mar**
# Refactored scope names
Most of the RBAC scopes are constructed in such a way that `<resource>:<subresource>` requires a subset of the permissions of `<resource>`. This PR fixes this for all scopes, renaming potentially confusing scope names such as `users:servers` (that is not actually a subscope of `users`) to `servers`. The documentation has been updated to reflect this. 

In addition, the generation of the scope description in the REST API is now generated from the source code.

### CODE REVIEW

**Comment by minrk** [comment]
We seem to have introduced a couple of flaky tests, but I don't think it's related to this PR, so going to go ahead and merge.

File: docs/rest-api.yml [hunk 1]
```diff
@@ -17,38 +17,53 @@ securityDefinitions:
...
+      all: Everything that the token-owning entity can access _(metascope for tokens)_
+      admin:users:
+        Read, write, create and delete users and their authentication state,
+        not including their servers or tokens.
+      admin:auth_state: Read a user’s authentication state.
+      users:
+        Read and write permissions to user models (excluding servers, tokens
+        and authentication state).
+      read:users:
+        Read user models (excluding including servers, tokens and authentication
```
Comment (manics) [inline docs/rest-api.yml:34]: `excluding` or  `including`?

File: docs/rest-api.yml [hunk 2]
```diff
@@ -700,7 +745,7 @@ paths:
       summary: Get a service by name
       security:
         - oauth2:
-            - read:services
+            - read:services - read:services:name - read:roles:services
```
Comment (manics) [inline docs/rest-api.yml:748]: Is this meant to be a list- if so it should be split over multiple lines.

**Review by manics — COMMENTED** [review]
@0mar @minrk I've spotted a couple of errors in the `rest-api.yml` doc

File: docs/rest-api.yml (hunk 2 above)
Comment (0mar) [reply docs/rest-api.yml:748]: Good catch, thanks!

File: docs/rest-api.yml (hunk 1 above)
Comment (0mar) [reply docs/rest-api.yml:34]: Excluding; access to servers, tokens and auth_state info is not granted with the `read:users` scope

**Pull request merged**